src/rpcrawtransaction.cpp

```diff
@@ -24,111 +23,280 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
+                std::string sProjectCommit = ExtractXML(msg, "<MA>", "</MA>");
+
+                if (sProjectCommit == "A")
+                    res.push_back(std::make_pair("Project Commit", "Add"));
+
+                else if (sProjectCommit == "D")
+                    res.push_back(std::make_pair("Project Commit", "Delete"));
+
+                else
+                    res.push_back(std::make_pair("Project Commit", "Unknown"));
+
+                res.push_back(std::make_pair("Project Name", sProjectName));
+                res.push_back(std::make_pair("Project URL", sProjectURL));
+            }
+            else
+            {
+                res.push_back(std::make_pair("Contract Type", "Unknown"));
+
+                if (fDebug)
+                    res.push_back(std::make_pair("Contract Data", msg));
+            }
+        }
+
+        else if (sTxMessage.length())
         {
-            res["IsSuperBlock"]=true;
+            res.push_back(std::make_pair("Message Type", "Transaction Message"));
+            res.push_back(std::make_pair("Message", sTxMessage));
         }
-    }
 
-}
+        else if (sRainMessage.length())
+        {
+            res.push_back(std::make_pair("Message Type", "Transaction Rain Message"));
+            res.push_back(std::make_pair("Message", sRainMessage));
+        }
 
-void GetTxNormalBoincHashInfo(json_spirit::mObject& res, const CMerkleTx& mtx)
-{
-    assert(!mtx.IsCoinStake() && !mtx.IsCoinBase());
-    res["blockhash"]=mtx.hashBlock.GetHex();
-    const std::string &hashBoinc = mtx.hashBoinc;
-    const std::string &msg = mtx.hashBoinc;
-
-    /* Possible formats:
-        * transaction <MESSAGE>
-        * a message <MT>
-          * cpid beacon
-          * vote
-          * poll
-        * unknown / text
-    */
+        else if (sMessageType.empty() && sTxMessage.empty() && sRainMessage.empty())
+            res.push_back(std::make_pair("Message Type", "No Attached Messages"));
 
-    res["bhLength"]=msg.length();
+        else
+        {
+            // Should never really make it here but if we do!
+            res.push_back(std::make_pair("Message Type", "Unknown Message"));
 
-    std::string sMessageType = ExtractXML(msg,"<MT>","</MT>");
-    std::string sTrxMessage = ExtractXML(msg,"<MESSAGE>","</MESSAGE>");
+            if (fDebug)
+                res.push_back(std::make_pair("HashBoinc", msg));
+        }
 
-    if(sMessageType.length())
-    {
-        res["bhType"]="message";
-        res["msgType"]=sMessageType;
+        return res;
     }
-    else if(sTrxMessage.length())
+
+    catch (const std::invalid_argument& e)
     {
-        res["bhType"]="TrxWithNote";
+        std::string sE(e.what()); // Pass through constructer to use as a string in error message
+        res.push_back(std::make_pair("ERROR", "Invalid argument exception while parsing Transaction Messages -> " + sE));
+        return res;
     }
-    else if(msg.length())
+
+    catch (const std::bad_alloc& e)
```

Comment:
bad_alloc should never be cought. It usually means pretty serious memory issue that is only safely treated by exit/crash.

Reply:
yea i'll remove it i had originally put it there to se what the cause was. turns out a bad beacon contract that cant decode in DecodeBase64 causing one and a seg fault i was trying to catch the location in my code before i reliazed it was not my code

Reply:
--complete